app/src/app/views/admin-page/user-accounts-page/user-accounts-page.component.ts

```diff
@@ -41,44 +47,49 @@ export class UserAccountsPageComponent implements OnInit {
             next: (data) => {
                 this.userAccounts = data.data;
-                this.filteredAccounts = data.data;
-                this.totalObjectCount = this.filteredAccounts.length;
-                this.paginator.pageIndex = 0;
+                this.totalObjectCount = data.pagination.total;
             },
-            complete: () => { this.userSubscription.unsubscribe() }
+            complete: () => {
+                this.userSubscription.unsubscribe()
+            }
         });
     }
 
-    public handleFilterSelection(selectedFilter): void {
-        this.selectedFilters = selectedFilter.value;
-        this.filteredAccounts = this.applyFilters(this.userAccounts);
+    public applyFilters(filters, applyControls = false): void {
+        let roleFilters = [];
+        let statusFilters = [];
+        let limit = this.paginator ? this.paginator.pageSize : 10;
+        let offset = this.paginator ? this.paginator.pageIndex * limit : 0;
+        if (!applyControls && this.paginator) this.paginator.pageIndex = 0;
+        this.selectedFilters = filters;
+        filters.forEach((filter) => {
+            if (this.filterOptions[0].values.includes(filter)) { // Status
+                statusFilters.push(filter)
+            } else if (this.filterOptions[1].values.includes(filter)) { // Role
+                roleFilters.push(filter)
+            }
+        })
+        this.getAccounts({limit: limit, offset: offset, status: statusFilters, role: roleFilters});
     }
 
-    public applyFilters(accounts): any {
-        let filtered = accounts;
-        if (this.selectedFilters.length > 0) {
-            const filters = this.filterOptions.map((f) => f.name.toLowerCase())
-            filtered = filtered.filter((acc) => filters.map((key) => acc[key]).some((val) => this.selectedFilters.indexOf(val) > -1))
-        }
-        return filtered;
+    public applySearch(query, applyControls = false): void {
+        let limit = this.paginator ? this.paginator.pageSize : 10;
+        let offset = this.paginator ? this.paginator.pageIndex * limit : 0;
+        if (!applyControls && this.paginator) this.paginator.pageIndex = 0;
+        this.getAccounts({limit: limit, offset: offset, search: query});
     }
 
-    public applySearch(query): void {
-        this.filteredAccounts = (this.userAccounts as any).filter((acc) => {
-            return Object.keys(acc).some((key) => {
-                return acc[key] && acc[key].includes(query);
-            });
-        });
-        this.filteredAccounts = this.applyFilters(this.filteredAccounts);
-    }
-
-    public applyControls(search?: string): void {
-        const limit = this.paginator ? this.paginator.pageSize : 10;
-        const offset = this.paginator ? this.paginator.pageIndex * limit : 0;
+    public applyControls(): void {
+        if (this.searchQuery) this.applySearch(this.searchQuery, true);
+        if (this.selectedFilters) this.applyFilters(this.selectedFilters, true);
     }
```

Comment:
The user accounts displayed in the list remain the same when clicking "next" or "back" on the paginator. The paginator's `page` event calls the `applyControls()` function - we would need a sort of "filter" here that shows the accounts within the currently selected page range.